test/integration/testUtil.js

```diff
@@ -140,6 +170,54 @@ module.exports.executeQueryAndVerify = function (connection, sql, expected, call
   connection.execute(executeOptions);
 };
 
+module.exports.executeQueryAndVerifyUsePool = function (connectionPool, sql, expected, callback, bindArray, normalize, strict)
+{
+  // Sometimes we may not want to normalize the row first
+  normalize = (typeof normalize !== "undefined" && normalize != null) ? normalize : true;
+  strict = (typeof strict !== "undefined" && strict != null) ? strict : true;
+  var executeOptions = {};
+  executeOptions.sqlText = sql;
+  executeOptions.complete = function (err, stmt)
+  {
+    assert.ok(!err, JSON.stringify(err));
+    var rowCount = 0;
+    var stream = stmt.streamRows();
+    stream.on('readable', function ()
+    {
+      var row;
+      while ((row = stream.read()) !== null)
+      {
+        if (strict)
+        {
+          assert.deepStrictEqual(normalize ? normalizeRowObject(row) : row, expected[rowCount]);
```

Comment:
if any assertion fails then callback won't be called at all?

Reply:
what callback does is to stop the execution. I think assertion failure will do the same thing.